src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1057,10 +1057,10 @@ internal OneOrMany<SyntaxTree> GetSyntaxTreesByMappedPath(string mappedPath)
             ImmutableSegmentedDictionary<string, OneOrMany<SyntaxTree>> computeMappedPathToSyntaxTree()
             {
                 var builder = ImmutableSegmentedDictionary.CreateBuilder<string, OneOrMany<SyntaxTree>>();
-                var resolver = Options.SourceReferenceResolver;
+                var resolver = Options.SourceReferenceResolver ?? SourceFileResolver.Default;
```

Comment:
I didn't quite follow this change. Do we have tests showing the impact of this?

Reply:
Added some tests which show the effect by mixing and matching directory separators.

Reply:
I worry a bit about this because it means that different parts of the compiler could end up using different resolvers. Did you consider just effectively making the resolver a non-nullable property? Essentially change options itself to return `SourceFileResolver.Default` if the current value is `null`? I can't really think of a case where `null` is a valid / interelting property here. 

Not 100% sold on doing this but wondering if you considered it here.

Reply:
It seems reasonable to me to have a non-null default. 99.9% of the time in real builds we have an instance. But in tests we don't have an instance unless we go out of our way to put one in. So, testing the weird combinations of test options which cause us to either have or not have a `SourceReferenceResolver` is just an annoyance.